From fcc68118f94675fd31c4458e4e5368a51a4adc26 Mon Sep 17 00:00:00 2001
From: Evgeni Golov <evgeni@golov.de>
Date: Mon, 7 Apr 2014 12:14:58 +0200
Subject: [PATCH] make building the driver optional

this allows package maintainers to build only the userspace binary
and let the user build the driver module for their specific kernel
which is not available in the build environment of the maintainer.
---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7fe880d..2d31ed6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,6 +48,8 @@ else()
 	message (STATUS "Using bundled jsoncpp in '${JSONCPP_SRC}'")
 endif()
 
+option(BUILD_DRIVER "Build the driver on Linux" ON)
+
 if(NOT WIN32)
 
 	set(SYSDIG_DEBUG_FLAGS "-D_DEBUG")
@@ -61,7 +63,7 @@ if(NOT WIN32)
 	set(CMAKE_C_FLAGS_RELEASE "-O3 -fno-strict-aliasing -DNDEBUG")
 	set(CMAKE_CXX_FLAGS_RELEASE "-O3 -fno-strict-aliasing -DNDEBUG")
 
-	if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+	if(CMAKE_SYSTEM_NAME MATCHES "Linux" AND BUILD_DRIVER)
 		add_subdirectory(driver)
 	endif()
 
-- 
1.9.1

